src/bci_build/package/versions.py

```diff
@@ -167,7 +166,18 @@ def format_version(ver: str, format: ParseVersion) -> str:
             raise ValueError(f"Invalid version format: {format}")
 
 
-async def update_versions(osc: Osc) -> dict[str, dict[str, str]]:
+def fetch_package_version(prj: str, pkg: str) -> str:
+    """Ask the OBS API to parse the source spec file version and return it."""
+    fetch = requests.get(
+        f"https://api.opensuse.org/public/source/{prj}/{pkg}",
+        params={"view": "info", "parse": "1"},
+        headers={"User-Agent": "BCI update-versions"},
+    )
+    fetch.raise_for_status()
+    return getattr(ET.fromstring(fetch.text).find("version"), "text", None)
```

Comment:
This means the type hint needs adjustment or raise an exception if `None` is returned

Reply:
true, I made it raise an exception